lib/routes/account.js

```diff
@@ -647,6 +714,99 @@ module.exports = function (
       }
     },
     {
+      method: 'POST',
+      path: '/account/device',
+      config: {
+        auth: {
+          strategy: 'sessionToken'
+        },
+        validate: {
+          payload: isA.object({
+            id: isA.string().length(32).regex(HEX_STRING).optional(),
+            name: isA.string().max(255).optional(),
+            type: isA.string().max(16).optional(),
+            pushCallback: isA.string().uri({ scheme: 'https' }).max(255).optional().allow(''),
+            pushPublicKey: isA.string().length(64).regex(HEX_STRING).optional().allow('')
+          })
+        },
+        response: {
+          schema: {
+            id: isA.string().length(32).regex(HEX_STRING).required(),
+            createdAt: isA.number().positive().optional(),
+            name: isA.string().max(255).optional(),
+            type: isA.string().max(16).optional(),
+            pushCallback: isA.string().uri({ scheme: 'https' }).max(255).optional().allow(''),
+            pushPublicKey: isA.string().length(64).regex(HEX_STRING).optional().allow('')
+          }
+        }
+      },
+      handler: function (request, reply) {
+        log.begin('Account.device', request)
+        var payload = request.payload
+        var sessionToken = request.auth.credentials
+        var operation = payload.id ? 'updateDevice' : 'createDevice'
+        db[operation](sessionToken.uid, sessionToken.tokenId, payload).then(
+          function (device) {
+            reply(butil.unbuffer(device))
+          },
+          reply
+        )
+      }
+    },
+    {
+      method: 'GET',
+      path: '/account/devices',
+      config: {
+        auth: {
+          strategy: 'sessionToken'
+        },
+        response: {
+          schema: isA.array().items(isA.object({
+            id: isA.string().length(32).regex(HEX_STRING).required(),
+            sessionToken: isA.string().length(64).regex(HEX_STRING).required(),
```

Comment:
@rfk, I nuked your comment when I pushed just now, so here it is again in case anyone wants to reply to it:

> I want to flag this for extra sanity-checking. We're now returning session-token ids that belong to other devices. I think this is OK, because you can't do anything with just the id, you also need to know the corresponding session key. But I may be missing something. Thoughts, anyone?